src/extensions/layout-selector/index.js

```diff
@@ -117,6 +140,80 @@ class LayoutSelector extends Component {
 		} );
 	}
 
+	detectImageBlocks( clientIds ) {
+		const {
+			getBlockName,
+			getBlockAttributes,
+		} = this.props;
+
+		const imageBlockTypes = [ 'core/image', 'core/cover' ];
+		const galleryBlockTypes = [ 'core/gallery' ];
+
+		return clientIds.map( ( clientId ) => {
+			const blockName = getBlockName( clientId );
+			const blockAttributes = getBlockAttributes( clientId );
+
+			if ( imageBlockTypes.includes( blockName ) ) {
+				return { [ clientId ]: pick( blockAttributes, [ 'id', 'url' ] ) };
+			}
+
+			if ( galleryBlockTypes.includes( blockName ) ) {
+				return { [ clientId ]: pick( blockAttributes, [ 'ids', 'images' ] ) };
+			}
+		} );
+	}
+
+	uploadExternalImages( clientId, blockAttributes ) {
+		const {
+			getBlockAttributes,
+			mediaUpload,
+			updateBlockAttributes,
+		} = this.props;
+
+		const urls = [];
+
+		if ( blockAttributes.hasOwnProperty( 'images' ) ) {
+			blockAttributes.images.forEach( ( image ) => urls.push( image.url ) );
+		} else {
+			urls.push( blockAttributes.url );
+		}
+
+		urls.forEach( ( imageUrl, index ) => {
+			window.fetch( imageUrl )
+				.then( ( response ) => response.blob() )
+				.then( ( blob ) => {
+					mediaUpload( {
+						filesList: [ blob ],
+						allowedTypes: [ 'image' ],
+						onFileChange( [ media ] ) {
+							if ( blockAttributes.hasOwnProperty( 'images' ) ) {
+								const currentAttributes = getBlockAttributes( clientId );
+
+								const newImages = currentAttributes.images.map( ( oldImage, oldIndex ) => {
+									return oldIndex === index
+										? Object.assign( {}, oldImage, { url: media.url } )
+										: oldImage;
+								} );
+
+								updateBlockAttributes( clientId, {
+									ids: media.id !== null ? [ ...currentAttributes.ids, media.id ] : currentAttributes.ids,
+									images: newImages,
+								} );
+							} else {
+								updateBlockAttributes( clientId, {
+									id: media.id,
+									url: media.url,
+								} );
+							}
+						},
+						onError( message ) {
+							console.log( [ 'onError', message ] );
```

Comment:
@jrtashjian Can we find a way to display an error to the end user directly in that case insteaf of burrying it inside the console?

Reply:
Agree that the errors need to be surfaced for the end-user. I attempted a number of URLs with images but am getting error `post.php?post=1985&action=edit:1 Access to fetch at 'https://www.cloudflare.com/img/learning/ddos/what-is-a-ddos-attack/syn-flood-ddos-attack.png' from origin 'https://wordpress.test' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.` followed by `Failed to load resource: net::ERR_FAILED`. The Cors issues may be unavoidable but if we have failure to upload, the user should be aware.

Reply:
I was afraid of this... not everything will be downloadable from the browser itself. Wondering if we need to come up with a way to download the image server side and then send it down to the front end to circumvent CORS issues.

Reply:
If remote files are fetched from a server we can control this is a non issue.

Reply:
Fixed in 30ff716. I used a warning notice over an error notice because the issue is non-blocking for the user, we only want to let the user know we were unable to pull down the remote image.